spark-etl/src/main/scala/geotrellis/spark/etl/Etl.scala

```diff
@@ -1,69 +1,69 @@
 import geotrellis.raster.reproject._
-import geotrellis.raster.resample.{ ResampleMethod, NearestNeighbor }
+import geotrellis.raster.resample.{NearestNeighbor, ResampleMethod}
 import geotrellis.raster.stitch.Stitcher
 import geotrellis.spark._
-import geotrellis.spark.io.index.KeyIndexMethod
 import geotrellis.spark.tiling._
 import geotrellis.spark.pyramid._
 import geotrellis.spark.tiling._
 import geotrellis.util._
 import geotrellis.vector._
+import geotrellis.spark.etl.config._
 
-import com.typesafe.scalalogging.slf4j.Logger
 import org.apache.spark._
 import org.apache.spark.rdd.RDD
 import org.slf4j.LoggerFactory
+import com.typesafe.scalalogging.slf4j.Logger
 
 import scala.reflect._
 import scala.reflect.runtime.universe._
 
 object Etl {
-  val defaultModules = Array(s3.S3Module, hadoop.HadoopModule, accumulo.AccumuloModule, cassandra.CassandraModule)
+  val defaultModules = Array(s3.S3Module, hadoop.HadoopModule, file.FileModule, accumulo.AccumuloModule, cassandra.CassandraModule)
 
   def ingest[
     I: Component[?, ProjectedExtent]: TypeTag: ? => TilerKeyMethods[I, K],
     K: SpatialComponent: Boundable: TypeTag,
     V <: CellGrid: TypeTag: Stitcher: (? => TileReprojectMethods[V]): (? => CropMethods[V]): (? => TileMergeMethods[V]): (? => TilePrototypeMethods[V])
   ](
-    args: Seq[String], keyIndexMethod: KeyIndexMethod[K], modules: Seq[TypedModule] = Etl.defaultModules
-  )(implicit sc: SparkContext) = {
+     args: Seq[String], modules: Seq[TypedModule] = Etl.defaultModules
+   )(implicit sc: SparkContext) = {
     implicit def classTagK = ClassTag(typeTag[K].mirror.runtimeClass(typeTag[K].tpe)).asInstanceOf[ClassTag[K]]
     implicit def classTagV = ClassTag(typeTag[V].mirror.runtimeClass(typeTag[V].tpe)).asInstanceOf[ClassTag[V]]
 
-    /* parse command line arguments */
-    val etl = Etl(args)
-    /* load source tiles using input module specified */
-    val sourceTiles = etl.load[I, V]
-    /* perform the reprojection and mosaicing step to fit tiles to LayoutScheme specified */
-    val (zoom, tiled) = etl.tile(sourceTiles)
-    /* save and optionally pyramid the mosaiced layer */
-    etl.save[K, V](LayerId(etl.conf.layerName(), zoom), tiled, keyIndexMethod)
+    EtlConf(args) foreach { conf =>
+      val job = EtlJob(conf)
+      /* parse command line arguments */
+      val etl = Etl(job, modules)
+      /* load source tiles using input module specified */
+      val sourceTiles = etl.load[I, V]
+      /* perform the reprojection and mosaicing step to fit tiles to LayoutScheme specified */
+      val (zoom, tiled) = etl.tile(sourceTiles)
+      /* save and optionally pyramid the mosaiced layer */
+      etl.save[K, V](LayerId(etl.input.name, zoom), tiled)
+    }
   }
 }
 
-case class Etl(args: Seq[String], @transient modules: Seq[TypedModule] = Etl.defaultModules) {
+case class Etl(@transient etlJob: EtlJob, @transient modules: Seq[TypedModule] = Etl.defaultModules) {
```

Comment:
Why are these marked as transient?

I think the consequence of this would be to have null values assigned to the public vals on serialization and deserialization.